src/react/chat-client-context.tsx

```diff
@@ -0,0 +1,36 @@
+import { ChatClient } from '@ably/chat';
+import React from 'react';
+
+// Just as in the Ably React context, we need to make sure we never create more than one Ably Chat React context.
+// This might happen when exporting a context directly from a module -
+// there's a risk of creating multiple instances of the same context
+// if there are misconfigurations in module bundler or package manager on the consumer side of Ably Chat Context.
+// This can lead to problems like having a Chat Client instance added
+// in one context, and then attempting to retrieve it from another different context.
+// This is why a single Ably Chat context is created and stored in the global state.
+const contextKey = Symbol.for('__ABLY_CHAT_CLIENT_CONTEXT__');
+
+interface GlobalThis {
+  [contextKey]?: React.Context<ChatClientContextValue>;
+}
+
+const globalObjectForContext: GlobalThis = typeof globalThis === 'undefined' ? {} : (globalThis as GlobalThis);
+
+// Ably chat context contains an object which stores all provider options indexed by provider id,
+// which is used to get options set by specific `ChatClientProvider` after calling `React.useContext`.
+export type ChatClientContextValue = Record<string, ChatClientContextProviderProps>;
```

Comment:
I'm not sure I can follow this explanation.

Is this to be able to have more than one single ChatClient in a react app? So I could have one called `"default"` and one called `"another"`? Related to the `chatId` from use-chat-client.ts?

Reply:
It's the same explanation provided in the ably-js react component for providing the ably context, it would cause the same issue for us, i.e. trying to reference a room that doesn't exist in the accessed context. I think the idea is that, unless you explicitly want a different context provided, then you must understand the above issue and be happy to override it. For most users, they don't want to have to think about this, but we also don't want to remove the option for all users.